### PR TITLE
Confirm reset password

### PR DESCRIPTION
**Pull request by helgabalashova**
#### Description of changes
Auth.confirmResetPassword API rewrite

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/providers/cognito/apis/confirmResetPassword.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { AuthValidationErrorCode } from '../../../errors/types/validation';
+import { assertServiceError } from '../../../errors/utils/assertServiceError';
+import { assertValidationError } from '../../../errors/utils/assertValidationError';
+import { ConfirmResetPasswordRequest } from '../../../types';
+import { confirmResetPasswordClient } from '../utils/clients/ConfirmResetPasswordClient';
+
+export async function confirmResetPassword(
+	confirmResetPasswordRequest: ConfirmResetPasswordRequest<CognitoConfirmResetPasswordOptions>
+): Promise<void> {
+	const username = confirmResetPasswordRequest.username;
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/confirmResetPassword.ts:17]: Nit: Would prefer to de-structure these all at the top & then run through the runtime validation.

File: packages/auth/src/providers/cognito/apis/confirmResetPassword.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		!!password,
+		AuthValidationErrorCode.EmptyConfirmResetPasswordNewPassword
+	);
+	const code = confirmResetPasswordRequest.confirmationCode;
+	assertValidationError(
+		!!code,
+		AuthValidationErrorCode.EmptyConfirmResetPasswordConfirmationCode
+	);
+	const config = Amplify.config;
+	await confirmResetPasswordClient({
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/confirmResetPassword.ts:33]: Nit: We can just return this promise directly

```suggestion

	return confirmResetPasswordClient({
```

File: packages/auth/src/providers/cognito/apis/confirmResetPassword.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		Username: username,
+		ConfirmationCode: code,
+		Password: password,
+		ClientMetadata: 
+			confirmResetPasswordRequest.options?.serviceOptions?.clientMetadata ??
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/confirmResetPassword.ts:38]: Another nit-pick, but I think I'd prefer to de-structure `options` with the rest of the params and do this logic up top

**Review by jimblanc — APPROVED** [review]
Looks good! Some nits

File: packages/auth/__tests__/providers/cognito/confirmResetPassword.test.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+import { AmplifyErrorString, Amplify } from '@aws-amplify/core';
+import { AuthError } from '../../../src/errors/AuthError';
+import { AuthValidationErrorCode } from '../../../src/errors/types/validation';
+import { confirmResetPassword } from '../../../src/providers/cognito';
+import { ConfirmForgotPasswordException } from '../../../src/providers/cognito/types/errors/service';
+import * as confirmResetPasswordClient from '../../../src/providers/cognito/utils/clients/ConfirmResetPasswordClient';
+import { authAPITestParams } from './testUtils/authApiTestParams';
+
+describe('ConfirmResetPassword API happy path cases', () => {
```
Comment (AllanZhengYP) [inline packages/auth/__tests__/providers/cognito/confirmResetPassword.test.ts:12]: Super nit: Usually we put `happy path` and `error path` in sub-suites.

**Review by AllanZhengYP — APPROVED** [review]
💯

**Pull request merged**